doc: Add sysbuild migration guide (#15845)
nrfconnect/sdk-nrf: doc/nrf/config_and_build/zephyr_samples_sysbuild.rst

Using Zephyr samples with sysbuild

> ##################################

Author: @MarekPieta @kapi-no could you check this for accuracy, if it will work or if there are restrictions on what network core image can be used for zephyr samples, etc.?

Reviewer: @dchat-nordic, @grochu, it seems that this page mostly focuses on the Radio/Network core image, so you should take a look at this content (as owners of the IPC radio firmware)

Using Zephyr samples with sysbuild
##################################

> With the introduction of sysbuild in |NCS| images for devices that have multiple core must now configure what images they need in order to be built successfully and run. For applications and samples in the |NCS| repository, this is handled automatically but for samples that are in the ``zephyr`` directory that come from upstream Zephyr, these images must be selected manually when building an image for nRF53 devices (for nRF52 as this is a single core device, no additional images are needed for Bluetooth samples).

Author: No clue on nrf54h

|                                                         |                         |           |          | for Bluetooth, :kconfig:option:`SB_CONFIG_NETCORE_IPC_RADIO_BT_RPC` enables nRF RPC serialization for Bluetooth, or   |
|                                                         |                         |           |          | :kconfig:option:`SB_CONFIG_NETCORE_IPC_RADIO_IEEE802154` enables spinel serialization for IEEE 802.15.4.              |
+---------------------------------------------------------+-------------------------+-----------+----------+-----------------------------------------------------------------------------------------------------------------------+
| :kconfig:option:`SB_CONFIG_NETCORE_NONE`                | No image                | -         | -        |                                                                                                                       |
+---------------------------------------------------------+-------------------------+-----------+----------+-----------------------------------------------------------------------------------------------------------------------+

> The default for thing53 is the ``empty_net_core`` sample application, for other nRF53 devices the default is no image.

Reviewer 2: ```suggestion
The default for Thingy:53 is the ``empty_net_core`` sample application.
The default for other nRF53 devices is having no image added to the build.
```

> When configuring an application, for example :zephyr_file:`zephyr/samples/bluetooth/peripheral_hr`, it must be configured with a supported network core image to work, for basic Bluetooth samples ``hci_ipc``, ``multiprotocol_rpmsg`` or ``ipc_radio`` can be used.
> The following shows how to build a sample with the ``hci_ipc`` network image selected:

Reviewer 2: ```suggestion
When configuring an application, such as :zephyr_file:`zephyr/samples/bluetooth/peripheral_hr`, you must configure it with a supported network core image to ensure proper functionality.
For basic Bluetooth samples, you can use `hci_ipc`, `multiprotocol_rpmsg`, or `ipc_radio`.

The following instructions demonstrate how to build a sample with the `hci_ipc` network image selected:
```
Not sure if what you are trying to say here is that if you need a supported network core image to combine with your app running on the app core, you can use one of the 3 basic bluetooth samples listed above.

.. tabs::

    .. group-tab:: west

        west build -b <board> -- -DSB_CONFIG_NETCORE_HCI_IPC=y

    .. group-tab:: cmake

        cmake -GNinja -DBOARD=<board> -DSB_CONFIG_NETCORE_HCI_IPC=y -DAPP_DIR=<app_dir> <path_to_zephyr/share/sysbuild>

> Upon building the project and flashing it, both the main application and selected network core images will be flashed to the device and the sample application will run as expected.

Reviewer 2: ```suggestion
When building and programming such a project, both the main application and the selected network core image will be programmed to the device and the sample application will run as expected.
```

The selected network core image can be just one, right? @nordicjm

| :kconfig:option:`SB_CONFIG_NETCORE_NONE`                | No image                | -         | -        |                                                                                                                       |
+---------------------------------------------------------+-------------------------+-----------+----------+-----------------------------------------------------------------------------------------------------------------------+

The default for Thingy:53 is the ``empty_net_core`` sample application.
The default for other nRF53 devices is having no image added to the build.

When configuring an application, such as :zephyr_file:`zephyr/samples/bluetooth/peripheral_hr`, you must configure it with a supported network core image to ensure proper functionality.
For basic Bluetooth samples, you can use `hci_ipc`, `multiprotocol_rpmsg`, or `ipc_radio`.
The following instructions demonstrate how to build a sample with the `hci_ipc` network image selected:

.. tabs::

    .. group-tab:: west

>         west build -b <board> -- -DSB_CONFIG_NETCORE_HCI_IPC=y

Author: Can you provide the command line options needed for using it with bluetooth samples, and 802.15.4 samples?
